### ISSUE TITLE
range over incoming eventually quits

### ISSUE THREAD

**Issue by andrewwatson** [jeffallen/mqtt#15]
I stole this chunk of code from one of your examples and then added a Println() call:

```
	// Receiver
	for m := range mqttClient.Incoming {
		fmt.Print(m.TopicName, "\t")
		m.Payload.WritePayload(os.Stdout)
		fmt.Println("\tr: ", m.Header.Retain)
	}

	fmt.Println("ran out of incoming")
```

and saw that if there weren't any messages for some period of time it would exit the for loop and my program would end.

Is this because of something on the mosquitto server closing my connection for being idle?

**Comment by jeffallen** [comment]
I would expect that to happen when the TCP session is broken for some
reason. Is is possible you are passing through a NAT which times out?

**Comment by andrewwatson** [comment]
Yes, it's quite possible.  I'm running this program on a Raspberry Pi in my office and connecting to a mosquitto running in AWS.

I'm trying to think of a better strategy for handling errors that may involve putting that range loop in another loop.

**Comment by andrewwatson** [comment]
Could we export the Done channel from the client so that my program could be notified that it's shutdown and reconnect?

**Comment by jeffallen** [comment]
I am open to pull requests.
